src/main/java/org/anarres/parallelgzip/ParallelGZIPOutputStream.java

```diff
@@ -189,7 +202,8 @@ public void write(byte[] b, int off, int len) throws IOException {
     private void submit() throws IOException {
         emitUntil(nthreads - 1);
         emitQueue.add(executor.submit(block));
-        block = new Block(/* block.index + 1 */);
+        block = blocks.isEmpty() ? new Block() : blocks.remove(blocks.size() - 1);
```

Comment:
I appreciate the memory locality effects of using the same block(s) for as long as possible; perhaps `ArrayDeque.removeLast()`, which forces us up to 1.6, but I think that's OK now.